src/ludwig.c

```diff
@@ -2170,7 +2170,8 @@ int ludwig_colloids_update(ludwig_t * ludwig) {
   else {
```

Comment:
I think what we can actually use here is:
```
...
else {
  /* Run halo on the target, and copy back all the data */
  lb_halo(ludwig->lb);
  lb_memcpy(ludwig->lb, tdpmemcpyDeviceToHost);
}
```
This is one of the copies that needs to be eliminated when we get to the colloids.